src/network/core/packet.cpp

```diff
@@ -48,7 +48,14 @@ Packet::Packet(NetworkSocketHandler *cs, size_t limit, size_t initial_read_size)
 Packet::Packet(NetworkSocketHandler *cs, PacketType type, size_t limit) : pos(0), limit(limit), cs(cs)
 {
 	/* Allocate space for the the size so we can write that in just before sending the packet. */
-	this->Send_uint16(0);
+	size_t size = sizeof(PacketSize);
```

Comment:
Instead of assuming `sizeof(PacketSize)` is also the encoded byte-length for our packet size, maybe we should add a simple wrapper function that does that for us. Something like `GetLengthByteSize()`.

There are several instances in this change that uses `sizeof(PacketSize)`, meaning: the encoded length of the packet size. Being a bit more explicit about this intention makes reading it a bit easier.

(in other words: I just spend 10 minutes trying to understand where this was coming from, to finally find out `PacketSize` is just a typing alias for `uint16_t` :D )

Reply:
I added something more explicit for packet size and type.